vkcubepp (KHR_display) fails to build since v1.3.213

Regressed by https://github.com/KhronosGroup/Vulkan-Headers/commit/33d4dd987fc8

```
$ cmake -DCUBE_WSI_SELECTION=DISPLAY -B /tmp/vulkan-tools_build
$ cmake --build /tmp/vulkan-tools_build
[...]
FAILED: cube/CMakeFiles/vkcubepp.dir/cube.cpp.o
/usr/bin/c++ -DVK_ENABLE_BETA_EXTENSIONS -DVK_USE_PLATFORM_DISPLAY_KHR -I/tmp/vulkan-tools_build/cube -I/tmp/vulkan-tools/cube -I/tmp/vulkan-tools_build/cube/.. -isystem /usr/local/include -Wall -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -fno-strict-aliasing -fno-builtin-memcmp -Werror -std=c++11 -fno-rtti -fvisibility=hidden -MD -MT cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -MF cube/CMakeFiles/vkcubepp.dir/cube.cpp.o.d -o cube/CMakeFiles/vkcubepp.dir/cube.cpp.o -c /tmp/vulkan-tools/cube/cube.cpp
cube/cube.cpp:2965:37: error: no viable conversion from 'typename ResultValueType<vk::DisplayPlaneCapabilitiesKHR>::type' (aka 'ResultValue<vk::DisplayPlaneCapabilitiesKHR>') to 'vk::DisplayPlaneCapabilitiesKHR'
    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found);
                                    ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/vulkan/vulkan_structs.hpp:26060:26: note: candidate constructor not viable: no known conversion from 'typename ResultValueType<vk::DisplayPlaneCapabilitiesKHR>::type' (aka 'ResultValue<vk::DisplayPlaneCapabilitiesKHR>') to 'vk::DisplayPlaneAlphaFlagsKHR' (aka 'Flags<vk::DisplayPlaneAlphaFlagBitsKHR>') for 1st argument
    VULKAN_HPP_CONSTEXPR DisplayPlaneCapabilitiesKHR( VULKAN_HPP_NAMESPACE::DisplayPlaneAlphaFlagsKHR supportedAlpha_ = {},
                         ^
/usr/include/vulkan/vulkan_structs.hpp:26081:26: note: candidate constructor not viable: no known conversion from 'typename ResultValueType<vk::DisplayPlaneCapabilitiesKHR>::type' (aka 'ResultValue<vk::DisplayPlaneCapabilitiesKHR>') to 'const vk::DisplayPlaneCapabilitiesKHR &' for 1st argument
    VULKAN_HPP_CONSTEXPR DisplayPlaneCapabilitiesKHR( DisplayPlaneCapabilitiesKHR const & rhs ) VULKAN_HPP_NOEXCEPT = default;
                         ^
/usr/include/vulkan/vulkan_structs.hpp:26083:5: note: candidate constructor not viable: no known conversion from 'typename ResultValueType<vk::DisplayPlaneCapabilitiesKHR>::type' (aka 'ResultValue<vk::DisplayPlaneCapabilitiesKHR>') to 'const VkDisplayPlaneCapabilitiesKHR &' for 1st argument
    DisplayPlaneCapabilitiesKHR( VkDisplayPlaneCapabilitiesKHR const & rhs ) VULKAN_HPP_NOEXCEPT
    ^
/usr/include/vulkan/vulkan.hpp:5917:5: note: candidate function
    operator std::tuple<Result &, T &>() VULKAN_HPP_NOEXCEPT
    ^
1 error generated.
```

FWIW, GCC is less verbose:
```c++
cube/cube.cpp: In member function 'vk::Result Demo::create_display_surface()':
cube/cube.cpp:2965:83: error: conversion from 'vk::ResultValueType<vk::DisplayPlaneCapabilitiesKHR>::type' {aka 'vk::ResultValue<vk::DisplayPlaneCapabilitiesKHR>'} to non-scalar type 'vk::DisplayPlaneCapabilitiesKHR' requested
 2965 |     vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found);
      |                                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -33,11 +33,14 @@ jobs:
                 cxx: [ g++, clang++ ]
                 config: [ Debug, Release ]
                 os: [ ubuntu-18.04, ubuntu-20.04 ]
+                cube_wsi: [ XCB, XLIB, WAYLAND, DISPLAY ]
                 exclude:
                     - cc: gcc
                       cxx: clang++
                     - cc: clang
                       cxx: g++
+                    - os: ubuntu-18.04
+                      cube_wsi: WAYLAND
 
         steps:
             - uses: actions/checkout@v2
@@ -52,7 +55,7 @@ jobs:
               run: python scripts/update_deps.py --dir external
 
             - name: Generate build files
-              run: cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=${{matrix.config}} -Cexternal/helper.cmake
+              run: cmake -S. -Bbuild -DCMAKE_BUILD_TYPE=${{matrix.config}} -DCUBE_WSI_SELECTION=${{matrix.cube_wsi}} -Cexternal/helper.cmake
               env:
                 CC: ${{matrix.cc}}
                 CXX: ${{matrix.cxx}}

--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -2962,7 +2962,7 @@ vk::Result Demo::create_display_surface() {
         exit(1);
     }
 
-    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found);
+    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found).value;
     // Find a supported alpha mode
     vk::DisplayPlaneAlphaFlagBitsKHR alphaMode = vk::DisplayPlaneAlphaFlagBitsKHR::eOpaque;
     std::array<vk::DisplayPlaneAlphaFlagBitsKHR, 4> alphaModes = {

